File: yh_tutorial/yh_tutorial_1/src/yh_sub_1.cpp
#include "ros/ros.h"
#include "yh_tutorial_1/yh_msg_1.h"

void msgCallback(const yh_tutorial_1::yh_msg_1::ConstPtr& msg)
{
    ROS_INFO("%d", msg->data);
}

int main(int argc, char ** argv)
{
    ros::init(argc, argv, "yh_sub_1");
    ros::NodeHandle nh;

    ros::Subscriber sub = nh.subscribe("yh_topic_1", 100, msgCallback);

    ros::spin();

    return 0;
}

File: yh_tutorial/yh_tutorial_1/scripts/yh_sub_1.py
#!/usr/bin/python
# -*- coding: utf8 -*-

from logging.config import listen
import rospy
from yh_tutorial_1.msg import yh_msg_1

def msgCallback(msg):
    rospy.loginfo("%d", msg.data)

def listener():
    rospy.init_node("yh_sub_1", anonymous=True)
    rospy.Subscriber("yh_topic_1", yh_msg_1, msgCallback)

    rospy.spin()

if __name__ == "__main__":
    listener()